pyo3-macros-backend/src/pyclass.rs

```diff
@@ -381,6 +381,65 @@ struct PyClassEnumVariant<'a> {
     /* currently have no more options */
 }
 
+struct PyClassEnum<'a> {
+    ident: &'a syn::Ident,
+    // The underyling representation of the enum.
+    // It's used to implement __int__ and __richcmp__.
+    // This matters when the underyling representation may not fit in `isize`.
+    repr: syn::Ident,
+    variants: Vec<PyClassEnumVariant<'a>>,
+    doc: PythonDoc,
+}
+
+impl<'a> PyClassEnum<'a> {
+    fn new(enum_: &'a syn::ItemEnum) -> syn::Result<Self> {
+        fn is_numeric_type(t: &syn::Ident) -> bool {
+            [
+                "u8", "i8", "u16", "i16", "u32", "i32", "u64", "i64", "u128", "i128", "usize",
+                "isize",
+            ]
+            .iter()
+            .any(|&s| t == s)
+        }
+        struct Reprs(syn::punctuated::Punctuated<syn::Ident, Token![,]>);
+        impl Parse for Reprs {
+            fn parse(input: ParseStream) -> Result<Self> {
+                let inner = Punctuated::parse_terminated(input)?;
+                Ok(Self(inner))
+            }
+        }
+        let ident = &enum_.ident;
+        // According to the [reference](https://doc.rust-lang.org/reference/items/enumerations.html),
+        // "Under the default representation, the specified discriminant is interpreted as an isize
+        // value", so `isize` should be enough by default.
+        let mut repr = syn::Ident::new("isize", proc_macro2::Span::call_site());
+        for attr in &enum_.attrs {
+            if attr.path.is_ident("repr") {
+                let reprs: Reprs = attr.parse_args()?;
+                for r in reprs.0 {
+                    if is_numeric_type(&r) {
+                        repr = r;
+                        break;
+                    }
+                }
+            }
```

Comment:
Looks like `reprs.0.into_iter().find()` could be used here, if we don't end up changing the parsing logic.